### PR TITLE
add targets for promoting RC artifacts

### PR DESCRIPTION
**Pull request by sdowell**
These targets are intended to be used for promoting release candidates.
The images are retagged from the original RC images. The manifests and
CLI binaries are rebuilt using the formal release tag.

### CODE REVIEW

**Comment by sdowell** [comment]
/hold
This change should rebase on https://github.com/GoogleContainerTools/kpt-config-sync/pull/47 to produce the multirepo/oss manifests

File: scripts/tag-promotion.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+read -rp "${RC_TAG} will be promoted to ${CS_VERSION} - Proceed (yes/no)? " choice
+case "${choice}" in
+  yes) ;;
+  no) exit 1 ;;
+  *) err "Unrecognized choice ${choice}"
+esac
+
+REMOTE="git@github.com:GoogleContainerTools/kpt-config-sync.git"
+# Use Github access token if provided
+[[ -n "${GH_TOKEN}" ]] && REMOTE="https://${GH_TOKEN}@github.com/GoogleContainerTools/kpt-config-sync.git"
```
Comment (nan-yu) [inline scripts/tag-promotion.sh:53]: Just curious, are you planning to configure GH_TOKEN in the release flow?

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**